### ISSUE TITLE
DateBox throws error with custom displayFormat

### ISSUE THREAD

**Issue by kvet** [DevExpress/DevExtreme-Intl#7]
Type text in DateBox with following options:
```js
$("#datebox-custom").dxDateBox({
    displayFormat: { weekday: 'long', month: 'short', day: 'numeric' },
    value: new Date()
});
```
You will see following error message: `Uncaught TypeError: format.toLowerCase is not a function`

**Comment by hneuer** [comment]
Hy!

If you enable a dataGrid's searchPanel and have a date column with a format as described above the same error occurs when you type a text into the searchPanel.

Is there an ETA for a solution of this bug? We use quite some datagrids in our application and most of them are having this problem now...

**Comment by nightskylark** [comment]
Hi
Thank you for the additional information. We are working on this issue and will publish a fix as soon as possible.

**Comment by DokaRus** [comment]
@hneuer 
As a workaround, you can define a format as an object with custom [formatter](https://js.devexpress.com/Documentation/16_2/ApiReference/Common/Object_Structures/format/#formatter) and [parser](https://js.devexpress.com/Documentation/16_2/ApiReference/Common/Object_Structures/format/#parser) methods, like in code below:

```
columns: [{
    dataField: "OrderDate",
    alignment: "right",
    dataType: "date",
    format: {
        formatter: function(value) {
            var format = { year: 'numeric', month: 'numeric', day: 'numeric' };
            return (new Intl.DateTimeFormat(DevExpress.localization.locale(), format)).format(value);
        },
        parser: function(string) {
            return new Date(string);
        }
    }
}]
```
 
At the moment, we think about the best way to improve the DevExtreme base localization module. As these changes will affect devextreme packages, I've created [the new ticket](https://www.devexpress.com/Support/Center/Question/Details/T473999) at our Support Center so you can subscribe to it.

**Comment by nightskylark** [comment]
Fixed in DevExtreme 16.2.5
